jose-emi/app.js

```diff
@@ -0,0 +1,323 @@
+var express = require('express');
+var app = express();
+var bodyParser = require('body-parser');
+
+app.use(bodyParser.json());
+
+var diasMiliseg = 1000 * 60 * 60 * 24
+
+function Libro(titulo,id,cantidad){
+    this.id = id;
+    this.titulo = titulo; 
+    this.cantidad = cantidad;
+}
+
+function Socio(nombre,id){
+    this.id = id;
+    this.nombre = nombre;
+}
+
+function Prestamo(id,idLibro,idSocio, fechaVto){
+    this.id = id;
+    this.idLibro = idLibro;
+    this.idSocio = idSocio;
+    this.fechaVto = fechaVto;
+    
+}
+
+var socios = new Array();
+var libros = new Array();
+var prestamos = new Array();
+
+// REQ1.1 Añadir libros
+app.post('/libros',function(req,res){
+    if(req.body.cantidad>0 && Number.isInteger(req.body.cantidad)==true){
+        var nuevoLibro = new Libro(req.body.titulo,getNewID(),req.body.cantidad);
+        libros.push(nuevoLibro);
+        res.status(201).json(nuevoLibro.id);
+    }
+    else{
+        res.status(400).json("Cantidad debe ser mayor a 0 y entera");
+    }
+})
+
+// REQ1.2 Eliminar libros
+app.delete('/libros/:idLibro',function(req,res){
+    var remover = buscarID(libros, req.params.idLibro);
+    if(remover != null){
+        var librosPrestados = buscarLibrosPrestados(prestamos);
+        if(!existeLibroPrestado(librosPrestados,req.params.idLibro)){
+            libros.splice(econtrarIndicePorId(libros, remover.id), 1);
+            res.status(204).json(remover.id);
+        }
+        else{
+            res.status(400).json("No se puede eliminar un libro prestado");
+        }
+    }
+    else{
+        // el parametro idLibro no existe en la coleccion LIBROS 
+        res.status(404).json("No existe el libro");
+    }   
+})
+
+// REQ1.3 Modificar cantidad de ejemplares de un libro
+app.put('/libros/:idLibro', function(req,res){
+    //validar cantidad valida
+    var libroMod;
+    if(Number.isInteger(req.body.cantidad) == true && req.body.cantidad > 0){
+        if(buscarID(libros, req.params.idLibro) != null)
+        {
+            if(req.body.cantidad>cantidadLibrosPrestados(req.params.idLibro)){
+                libroMod = modifCantEjemplares(req.params.idLibro, req.body.cantidad);
+                res.status(200).json(libroMod);
+            }
+            else{
+                res.status(400).json("La cantidad a ingresar es mayor a la cantidad prestada");
```

Comment:
Here the message should be "La cantidad a ingresar debe ser mayor a la cantidad prestada"